optuna/integration/sklearn.py

```diff
@@ -666,7 +668,7 @@ def __init__(
         param_distributions: Mapping[str, distributions.BaseDistribution],
         cv: Optional[Union["BaseCrossValidator", int]] = 5,
         enable_pruning: bool = False,
-        error_score: Union[Number, str] = np.nan,
+        error_score: Union[Number, float, str] = np.nan,
```

Comment:
`isinstance(np.nan, Number)` should be `True` but is this change still required?